apps/full-site-editing/full-site-editing-plugin/global-styles/src/dom-updater.js

```diff
@@ -14,23 +15,32 @@ export default ( options, getOptionValue ) => {
 	domReady( () => {
 		const current = {};
 		const cssVariables = {};
+		const styleElement = document.createElement( 'style' );
+		document.body.appendChild( styleElement );
+		const styleSheet = styleElement.sheet;
+
 		options.forEach( option => {
 			current[ option ] = null;
 			cssVariables[ option ] = `--${ option.replace( '_', '-' ) }`;
 		} );
 
 		subscribe( () => {
+			let styleProps = '';
 			Object.keys( current ).forEach( key => {
 				const value = getOptionValue( key );
-				if ( current[ key ] !== value ) {
+				if ( ! isEmpty( value ) && current[ key ] !== value ) {
```

Comment:
Just checking out loud the reasoning here: before this change, we actually needed empty/null values to go through this as that was the way to delete them from the node. With the new approach, though, we exclude those from the style string.

Reply:
I definitely might be missing something here but based on my observation in the current version of the code they were not deleted but set back to the default `unset`. I added the check because initially, the code runs a couple of times with `undefined` values. You can try for yourself by just logging the value of `value`. It's logging `undefined` twice at the start and then sets the default to `unset`. `unset` is also the value it's set to when setting it back to the Theme Default.

Reply:
If the value of `setProperty` is `null` it actually deletes the declaration (what I was thinking) but if it is `undefined` it remains as it is (TIL).

Reply:
Actually, to make things more interesting, the behavior of `undefined` changes by the value type/property:

* `setProperty('color', undefined)` => the value remains as it was
* `setProperty('font-family', undefined)` => it actually sets undefined as font-family

while null removes the declaration from the ruleset (`setProperty('color', null)` or `setProperty('font-family', null)` have the same behaviour).

Reply:
As per discussion on Slack future-proofed the implementation to account for `null` values.